staging/src/k8s.io/client-go/tools/events/eventseries_test.go

```diff
@@ -206,3 +206,161 @@ func validateEventSerie(messagePrefix string, expectedUpdate bool, actualEvent *
+	go eventBroadcaster.finishSeries()
+	actualEvent := <-patchEvent
+	t.Logf("validating event affected by patch request")
+	eventBroadcaster.mu.Lock()
+	if len(cache) != 0 {
+		t.Errorf("cache should be empty, but instead got a size of %v", len(cache))
+	}
+	eventBroadcaster.mu.Unlock()
+	if !actualEvent.Series.LastObservedTime.Equal(&cachedEvent.Series.LastObservedTime) {
+		t.Errorf("series was expected be seen with LastObservedTime %v, but instead got %v ", cachedEvent.Series.LastObservedTime, actualEvent.Series.LastObservedTime)
+	}
+}
+
+func refreshExistingEventSeries(t *testing.T) {
+	hostname, _ := os.Hostname()
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			SelfLink:  "/api/version/pods/foo",
+			Name:      "foo",
+			Namespace: "baz",
+			UID:       "bar",
+		},
+	}
+	regarding, err := ref.GetPartialReference(scheme.Scheme, testPod, ".spec.containers[1]")
+	if err != nil {
+		t.Fatal(err)
+	}
+	related, err := ref.GetPartialReference(scheme.Scheme, testPod, ".spec.containers[0]")
+	if err != nil {
+		t.Fatal(err)
+	}
+	LastObservedTime := metav1.MicroTime{Time: time.Now().Add(-9 * time.Minute)}
+
+	cachedEvent := &v1beta1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: "baz",
+		},
+		EventTime:           metav1.MicroTime{time.Now()},
+		ReportingController: "eventTest",
+		ReportingInstance:   "eventTest-" + hostname,
+		Action:              "started",
+		Reason:              "test",
+		Regarding:           *regarding,
+		Related:             related,
+		Note:                "some verbose message: 1",
+		Series: &v1beta1.EventSeries{
+			Count:            10,
+			LastObservedTime: LastObservedTime,
+		},
+		Type: v1.EventTypeNormal,
+	}
+
+	anotherCachedEvent := cachedEvent.DeepCopy()
+	anotherCachedEvent.ReportingInstance = "anotherEventTest"
+	createEvent := make(chan *v1beta1.Event)
+	updateEvent := make(chan *v1beta1.Event)
+	patchEvent := make(chan *v1beta1.Event)
+	testEvents := testEventSeriesSink{
+		OnCreate: func(event *v1beta1.Event) (*v1beta1.Event, error) {
+			createEvent <- event
+			return event, nil
+		},
+		OnUpdate: func(event *v1beta1.Event) (*v1beta1.Event, error) {
+			updateEvent <- event
+			return event, nil
+		},
+		OnPatch: func(event *v1beta1.Event, patch []byte) (*v1beta1.Event, error) {
+			// event we receive is already patched, usually the sink uses it only to retrieve the name and namespace, here
+			// we'll use it directly
+			patchEvent <- event
+			return event, nil
+		},
+	}
+	cacheKey := getKey(cachedEvent)
+	cache := map[eventKey]*v1beta1.Event{
+		cacheKey: cachedEvent,
+	}
+
+	eventBroadcaster := newBroadcaster(&testEvents, 0, cache).(*eventBroadcasterImpl)
+	go eventBroadcaster.refreshExistingEventSeries()
+	<-patchEvent
+	t.Logf("validating event affected by patch request")
+	eventBroadcaster.mu.Lock()
+	if len(cache) == 1 {
+		t.Errorf("cache should not be empty, but instead got a size of %v", len(cache))
+	}
+	eventBroadcaster.mu.Unlock()
+}
```

Comment:
Also remove the corresponding TODOs from the implementation file to add tests for these.

Reply:
Ack